notebooks/tfx_pipelines/pipeline/solutions/pipeline_vertex/model.py

```diff
@@ -0,0 +1,313 @@
+    Returns:
+      A namedtuple contains the following:
+        - tuner: A BaseTuner that will be used for tuning.
+        - fit_kwargs: Args to pass to tuner's run_trial function for fitting
+                      the model , e.g., the training and validation dataset.
+                      Required args depend on the above tuner's implementation.
+    """
+    transform_graph = tft.TFTransformOutput(fn_args.transform_graph_path)
+
+    # Construct a build_keras_model_fn that just takes hyperparams from
+    # get_hyperparameters as input.
+    build_keras_model_fn = functools.partial(
+        _build_keras_model, tf_transform_output=transform_graph
+    )
+
+    # BayesianOptimization is a subclass of kerastuner.Tuner which inherits
+    # from BaseTuner.
+    tuner = kerastuner.BayesianOptimization(
+        build_keras_model_fn,
+        max_trials=10,
+        hyperparameters=_get_hyperparameters(),
+        # New entries allowed for n_units hyperparameter construction
+        # conditional on n_layers selected.
+        #       allow_new_entries=True,
+        #       tune_new_entries=True,
+        objective=kerastuner.Objective(
+            "val_sparse_categorical_accuracy", "max"
+        ),
+        directory=fn_args.working_dir,
+        project_name="covertype_tuning",
+    )
+
+    train_dataset = _input_fn(
+        fn_args.train_files,
+        fn_args.data_accessor,
+        transform_graph,
+        batch_size=TRAIN_BATCH_SIZE,
+    )
+
+    eval_dataset = _input_fn(
+        fn_args.eval_files,
+        fn_args.data_accessor,
+        transform_graph,
+        batch_size=EVAL_BATCH_SIZE,
+    )
+
+    return TunerFnResult(
+        tuner=tuner,
+        fit_kwargs={
+            "x": train_dataset,
+            "validation_data": eval_dataset,
+            "steps_per_epoch": fn_args.train_steps,
+            "validation_steps": fn_args.eval_steps,
+        },
+    )
+
+
+# TFX Trainer will call this function.
+def run_fn(fn_args: TrainerFnArgs):
+    """Train the model based on given args.
+    Args:
+      fn_args: Holds args used to train the model as name/value pairs.
+    """
+
+    tf_transform_output = tft.TFTransformOutput(fn_args.transform_output)
+
+    train_dataset = _input_fn(
+        fn_args.train_files,
+        fn_args.data_accessor,
+        tf_transform_output,
+        TRAIN_BATCH_SIZE,
+    )
+
+    eval_dataset = _input_fn(
+        fn_args.eval_files,
+        fn_args.data_accessor,
+        tf_transform_output,
+        EVAL_BATCH_SIZE,
+    )
+
+    if fn_args.hyperparameters:
+        hparams = kerastuner.HyperParameters.from_config(
+            fn_args.hyperparameters
+        )
+    else:
+        # This is a shown case when hyperparameters is decided and Tuner is
+        # removed from the pipeline. User can also inline the hyperparameters
+        # directly in _build_keras_model.
+        hparams = _get_hyperparameters()
```

Comment:
You can just keep the _get_hyperparameters for the defaults and remove the keras tuner logic

Reply:
Done!